### PR TITLE
Disable failing compression test on OSX High Sierra

### PR DESCRIPTION
**Pull request by ianhays**
Port https://github.com/dotnet/corefx/pull/23276 to release/2.0.0

resolves https://github.com/dotnet/corefx/issues/21427

### CODE REVIEW

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs [hunk 1]
```diff
@@ -15,6 +15,8 @@ namespace System.IO.Compression.Tests
 {
     public partial class ZipFileTest_ConvenienceMethods : ZipFileTestBase
     {
+        public static bool IsNotMacOsHighSierraOrHigher => PlatformDetection.IsNotMacOsHighSierraOrHigher;
```
Comment (tarekgh) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: why you did that?

Comment (ianhays) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: There doesn't seem to be the type overload of `ConditionalFact` that allows you to specify a conditional function from a different class for the version of buildtools being used in rel/2.0.0

Comment (ianhays) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: Ah, yeah, that makes sense. The type name `ConditionalFact` was only added 11 days ago: https://github.com/dotnet/buildtools/pull/1634 and rel/2.0.0 doesn't have that change.

Comment (tarekgh) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: yes you need to do 

```
nameof(PlatformDetection)  + "." + nameof(PlatformDetection.IsNotMacOsHighSierraOrHigher)
```

in the ConditionalFact attribute

**Pull request merged**